### PR TITLE
feat(DTFS2-6199): NDB deal page TFM

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Added new summary for NDB with rearranged columns and without deal stage
* Removed eligibility criteria if NDB
* Added component tests

### CODE REVIEW

File: trade-finance-manager-ui/templates/case/case.njk [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
-  {{caseSummary.render(componentData)}}
+    {{caseSummaryNdb.render(componentData)}}
+
+  {% else %}
+    {% set componentData = {
+      deal: deal,
+      tfm: tfm,
+      user: user,
+      amendment: amendment
+    } %}
```
Comment (ttbarnes) [inline trade-finance-manager-ui/templates/case/case.njk:26]: Could this `set componentData` outside of the conditional? DRY

Comment (Zainzzkk) [reply trade-finance-manager-ui/templates/case/case.njk:26]: Your request has been fulfilled :P Just waiting for any @abhi-markan comments and then will push this change

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.component-test.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    it('should NOT add `chevron-right` class to supplier column', () => {
+      wrapper.expectElement('[data-cy="supplier-column"]').hasClass('case-summary-supplier');
+    });
+  });
+
+  describe('tier 1 exporter', () => {
+    beforeEach(() => {
+      params = JSON.parse(JSON.stringify(params));
+      params.deal.submissionDetails.supplierType = 'UK Supplier';
```
Comment (abhi-markan) [inline trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.component-test.js:140]: Can we feed it from a `Constant` please?

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        <div class="case-summary-overview-item">
+          <div class="ukef-heading-xs">Bank</div>
+          <div class="ukef-body-s" data-cy="bank-name">{{ deal.bank.name | dashIfEmpty }}</div>
+        </div>
+
+        <div class="case-summary-overview-item">
+          <div class="ukef-heading-xs">Date received</div>
+            <div class="ukef-body-s" data-cy="date-received">
+              {% if tfm.dateReceived %}
```
Comment (abhi-markan) [inline trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:52]: Can we use `dashIfEmpty` plugin here please?

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+          <div class="ukef-body-s">
+            <span data-cy="contract-value">
+              {% if deal.submissionDetails.supplyContractCurrency and deal.submissionDetails.supplyContractValue %}
+                {{ deal.submissionDetails.supplyContractCurrency }} {{ deal.submissionDetails.supplyContractValue | formatAsCurrency }}
+                {% else %}
+                -
+              {% endif %}
+            </span>
+            {% if tfm.supplyContractValueInGBP %}
+              <span data-cy="contract-value-in-gbp">{{ tfm.supplyContractValueInGBP | dashIfEmpty }}</span>
```
Comment (abhi-markan) [inline trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:97]: Do we need `formatAsCurrency`?

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            {% endif %}
+          </div>
+        </div>
+
+        <div class="case-summary-overview-item">
+          <div class="ukef-heading-xs">Total facilities in this deal</div>
+          <div class="ukef-body-s" data-cy="total-facilities-in-gbp">{{ deal.totals.facilitiesValueInGBP | dashIfEmpty }}</div>
```
Comment (abhi-markan) [inline trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:104]: Do we need formatAsCurrency?

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        </div>
+
+        <div class="case-summary-overview-item">
+          <div class="ukef-heading-xs">UKEF exposure for this deal</div>
+          <div class="ukef-body-s" data-cy="total-ukef-exposure">{{ deal.totals.facilitiesUkefExposure | dashIfEmpty }}</div>
```
Comment (abhi-markan) [inline trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:109]: Do we need formatAsCurrency?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Brilliant work, few comments to address please.

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk (hunk 3 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk]: done

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk (hunk 4 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:89]: done

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk (hunk 5 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:96]: done

File: trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk (hunk 6 above)
Comment (Zainzzkk) [reply trade-finance-manager-ui/templates/case/_macros/case-summary-ndb.njk:101]: done

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**